.github/workflows/integration.yaml

```diff
@@ -0,0 +1,138 @@
+# reusable workflow triggered by other actions
+name: Continuous Integration
+
+on:
+  workflow_call:
+    secrets:
+      charmcraft-credentials:
+        required: trueon:
+  - push
+  - pull_request
```

Comment:
I think the second `on` should be removed, because the CI is already triggered by `on_push.yaml` and `on_pull_request.yaml`.